src/WormholeConstantFee.sol

```diff
@@ -26,7 +26,7 @@ contract WormholeConstantFee is WormholeFees {
         fee = _fee;
     }
 
-    function getFees(WormholeGUID calldata, uint256, int256) override external view returns (uint256) {
-        return fee;
+    function getFees(WormholeGUID calldata guid, uint256, int256, uint256 amtToTake) override external view returns (uint256) {
+        return fee * amtToTake / guid.amount;
```

Comment:
Should handle the case where `guid.amount` is 0 and return 0 fees.

Reply:
Done.